source/wp-content/themes/wporg-developer-2023/src/dashicons-page/view.js

```diff
@@ -0,0 +1,104 @@
+import * as IAPI from '@wordpress/interactivity';
+
+const config = IAPI.getConfig( 'wporg/developer/dashicons-page' );
+const { state } = IAPI.store( 'wporg/developer/dashicons-page', {
+	state: {
+		get iconClass() {
+			return `dashicons ${ IAPI.getContext().icon }`;
+		},
+
+		get sectionAnchorTarget() {
+			return `icons-${ IAPI.getContext().section.slug }`;
+		},
+
+		get sectionAnchorHref() {
+			return `#icons-${ IAPI.getContext().section.slug }`;
+		},
+
+		get iconSectionLabel() {
+			return IAPI.getContext().section.label;
+		},
+
+		get eachIcon() {
+			return config.icons[ IAPI.getContext().icon ];
+		},
+
+		get style() {
+			const cleanedValue = state.filter.replace( /[^a-zA-Z0-9]/, '' );
```

Comment:
This only replaces the first matching instance, so `s.ite` works but `s.i.t.e` doesn't.

I would also suggest allowing `-`, since some icons use that as part of the name. For example, [menu-alt](https://developer.wordpress.org/resource/dashicons/?icon=dashicons-menu-alt) can't be found by searching "menu-alt".

```suggestion
			const cleanedValue = state.filter.replace( /[^a-zA-Z0-9-]/g, '' );
```